be/src/util/threadpool.cpp

```diff
@@ -255,6 +255,11 @@ ThreadPool::~ThreadPool() noexcept {
     CHECK_EQ(1, _tokens.size()) << strings::Substitute("Threadpool $0 destroyed with $1 allocated tokens", _name,
                                                        _tokens.size());
     shutdown();
+    // wait all sub threads done to release all threads (maybe from the main thread)
+    for (auto it = _pthread_ids.begin(); it != _pthread_ids.end(); it++) {
+        pthread_join(*it, nullptr);
+    }
+    _pthread_ids.clear();
```

Comment:
If you want to wait all threads quit before the the process exit, why don't you just call `ThreadPool::wait()`?

Reply:
`ThreadPool::~ThreadPool()` will call `ThreadPool::shutdown()` and `ThreadPool::shutdown()` will wait for the running tasks to complete, why do we need join here?

Reply:
I got it, you want to wait for all threads destroyed.